### PR TITLE
Set MSRV to 1.29 and fix README & Travis accordingly

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,30 +3,12 @@ rust:
   - stable
   - beta
   - nightly
-  - 1.24.0
+  - 1.29.0
 env:
   - BITCOINVERSION=0.18.0
   - BITCOINVERSION=0.18.1
   - BITCOINVERSION=0.19.0.1
   - BITCOINVERSION=0.19.1
```
Comment (RCasatta) [inline .travis.yml:11]: we already know `BITCOINVERSION=0.20.1` it's not compatible or we should add it?

Comment (RCasatta) [reply .travis.yml:11]: it may be out of scope...

Comment (stevenroose) [reply .travis.yml:11]: There's another MR for that. That's kinda why I did this one right now.

Comment (RCasatta) [reply .travis.yml:11]: sorry, I missed that

**Comment by RCasatta** [comment]
ack 1b759cbcd463e23f169f5e9ddd526fc0b01688b0

**Pull request merged**